new-lamassu-admin/src/pages/Dashboard/SystemPerformance/SystemPerformance.js

```diff
@@ -45,25 +60,47 @@ const GET_DATA = gql`
       dispense
       sendConfirmed
     }
-    fiatRates {
-      code
-      name
-      rate
-    }
-    config
   }
 `
 
 const SystemPerformance = () => {
   const classes = useStyles()
   const [selectedRange, setSelectedRange] = useState('Day')
-  const { data, loading } = useQuery(GET_DATA, {
-    variables: { excludeTestingCustomers: true }
+  const [transactions, setTransactions] = useState([])
+  const { data, loading: dataLoading } = useQuery(GET_DATA)
+  const NOW = Date.now()
+
+  const { loading: transactionsLoading } = useQuery(GET_TRANSACTION, {
+    variables: {
+      from: endOfMinute(subDays(31, NOW)),
+      excludeTestingCustomers: true
+    },
+    onCompleted: data => setTransactions(data?.transactions || [])
   })
-  const fiatLocale = fromNamespace('locale')(data?.config).fiatCurrency
-  const timezone = fromNamespace('locale')(data?.config).timezone
 
-  const NOW = Date.now()
+  useQuery(GET_TRANSACTION, {
+    variables: {
+      from: endOfMinute(subMinutes(3, NOW)),
+      excludeTestingCustomers: true
+    },
+    pollInterval: 10000,
+    notifyOnNetworkStatusChange: true,
+    onCompleted: data => {
+      setTransactions(
+        R.unionWith(
+          R.eqBy(R.prop('id')),
+          data?.transactions || [],
+          transactions
+        )
+      )
+    }
+  })
```

Comment:
Isn't it possible to combine these two queries into a single one?

```
const { data: latestTxs } = useQuery(GET_TRANSACTION, {
  variables: {
    from: R.isEmpty(transactions) ? endOfMinute(subDays(31, NOW)) : new Date(Date.now() - 10000),
    excludeTestingCustomers: true
  },
  pollInterval: 10000,
  notifyOnNetworkStatusChange: true,
  onCompleted: data => setTransactions(R.unionWith(R.eqBy(R.prop('id')), data.transactions || [], transactions))
})
```

If you want to know the initial loading flag to maintain the same behavior, probably best to use `useState`

```
const [initialTxLoading, setInitialTxLoading] = useState(false)

(...)

useQuery(GET_TRANSACTION, {
  variables: {
    from: R.isEmpty(transactions) ? endOfMinute(subDays(31, NOW)) : new Date(Date.now() - 10000),
    excludeTestingCustomers: true
  },
  pollInterval: 10000,
  notifyOnNetworkStatusChange: true,
  onCompleted: data => {
    setInitialTxLoading(true) // initialTxLoading can then be used to ensure that nothing is rendered without any transactions
    setTransactions(R.unionWith(R.eqBy(R.prop('id')), data.transactions || [], transactions))
  }
})
```